framework/src/problems/FEProblemBase.C

```diff
@@ -3884,7 +3884,7 @@ FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
     // non-nodal user objects have to be run separately before the nodal user objects run
     // because some nodal user objects (NodalNormal related) depend on elemental user objects :-(
     ComputeUserObjectsThread cppt(*this, getNonlinearSystemBase(), query);
-    Threads::parallel_reduce(*_mesh.getActiveLocalElementRange(), cppt);
+    Threads::parallel_reduce(getEvaluableElementRange(), cppt);
```

Comment:
This change is a no-go. This is going to mess with many calculations. As an example, you can imagine a elemental user object that computes a total domain volume will have the wrong result after this change because elements that are evaluable on multiple processes will multiply add their individual element volume. I can see the change to use evaluable elements for stateful material properties (`ComputeMaterialsObjectThread`), but that's the only loop I can see the change for.